heroku-deploy/src/main/java/com/heroku/sdk/deploy/lib/sourceblob/SourceBlobPackager.java

```diff
@@ -0,0 +1,69 @@
+package com.heroku.sdk.deploy.lib.sourceblob;
+
+import com.heroku.sdk.deploy.lib.OutputAdapter;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveOutputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorOutputStream;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public final class SourceBlobPackager {
+
+    public static Path pack(SourceBlobDescriptor sourceBlobDescriptor, OutputAdapter outputAdapter) throws IOException {
+        Path tarFilePath = Files.createTempFile("heroku-deploy", "source-blob.tgz");
+
+        TarArchiveOutputStream tarArchiveOutputStream = new TarArchiveOutputStream(
+                new GzipCompressorOutputStream(new FileOutputStream(tarFilePath.toFile())));
+
+        tarArchiveOutputStream.setLongFileMode(TarArchiveOutputStream.LONGFILE_GNU);
+
+        outputAdapter.logInfo("-----> Packaging application...");
```

Comment:
You don't have to keep the `----->`, and a major version release would be a good time to change if you want. That said, it still aligns well with the buildpack output.

Reply:
I will keep it as-is, since I have no better proposal.